src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -2011,5 +2011,59 @@ private static bool TryFindFirstMatchedLane(Vector128<byte> mask, Vector128<byte
             matchedLane = BitOperations.TrailingZeroCount(selectedLanes) >> 2;
             return true;
         }
+
+        public static void ReverseByteRef(ref byte buf, nint length)
+        {
+            ref byte first = ref buf;
+            ref byte last = ref Unsafe.Add(ref Unsafe.Add(ref first, length), -1);
+            int numBytesWritten = 0;
+            if (Avx2.IsSupported && Vector256<byte>.Count * 2 <= length)
+            {
+                Vector256<byte> ReverseMask = Vector256.Create((byte)0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, // first 128-bit lane
+                0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15);  // second 128-bit lane
+                last = ref Unsafe.Add(ref Unsafe.Add(ref first, length), -Vector256<byte>.Count);
+                do
+                {
+                    Vector256<byte> tempFirst = Unsafe.As<byte, Vector256<byte>>(ref first);
+                    Vector256<byte> tempLast = Unsafe.As<byte, Vector256<byte>>(ref last);
+                    tempFirst = Avx2.Shuffle(tempFirst, ReverseMask);
```

Comment:
Can we avoid Avx API and use only VectorN throughout? Then we benefit on Arm64 at the same time.

Reply:
I don't think so because System.Numerics.Vector doesn't have a portable shuffle implementation, to my knowledge.

Reply:
Α portable shuffle implementation will come soon as tannergooding said.

Reply:
@tannergooding, how close are we to having the Vector128/256 shuffle operation?  Wondering if we should block this PR on that, or just add this to the list of things to be fixed up subsequently.

Reply:
Its going to be a more complex PR to make happen since it requires recognizing several patterns and optimizing them appropriately.

I hope to have it in .NET 7 P2, but it might be closer to P3 depending on my own availability and the availability of the JIT side reviewers.

It's probably not worth blocking here and we can use this as one of the early diffs to ensure Shuffle is working as intended.